src/app/settings/admin/presets/component.ts

```diff
@@ -102,11 +106,33 @@ export class PresetListComponent implements OnInit, OnDestroy, OnChanges {
     };
 
     merge(of(true), this._presetsChanged)
-      .pipe(switchMap(_ => this._presets$))
-      .pipe(takeUntil(this._unsubscribe))
-      .subscribe(presetList => {
-        this.presets = presetList.items;
+      .pipe(
+        switchMap(_ => this._presets$),
+        tap((presetList: PresetList) => {
+          this.presets = presetList.items;
+        }),
+        switchMap(_ => {
+          const presetStats$ = [];
+          this.presets.forEach((preset: Preset) => {
+            presetStats$.push(this._presetService.getPresetStatsBy(preset.name));
+          });
+          this.isBusyCounter++;
+          return forkJoin(presetStats$);
```

Comment:
Fork join will wait until all subscriptions have emitted values AFAIR, why using counter if that is true?

Reply:
Definitely, the API in this case is not optimal. This will put a lot of overhead on the UI and API since for every preset on the list we will make a request for the stats. I'd vote for updating the API and not using it this way.

Reply:
@floreks that's correct. This is overhead on UI to have n+1 call for each preset and as no of presets will increase it will cause slight delays for page to show presets data + additional info on table. 

- +1 if it can be added up API `/presets` endpoint than from FE the extra API call be be removed.

Reply:
@maciaszczykm correct on first part.

why using counter if that is true?

Reason: on legacy code the check was `!presets` to show busy counter and it will be always true as `!! []` -> true so loader doesn't appear.

Adding **isBusyCounter**: number -> counter incremented when any API call is initiated so that spinner can be shown and disable spinner on when all api calls or task is been done until we reaches final subscription close the spinner by decrement **isBusyCounter**

thanks

Reply:
Hmm, okay. Let's change the API part if we can and then we can keep it. Would be nice to add it to error handler or finalizer too.

Reply:
Good Point 👍🏽